driver-core/src/main/com/mongodb/client/model/expressions/MapExpression.java

```diff
@@ -21,45 +21,162 @@
 import static com.mongodb.client.model.expressions.Expressions.of;
 import static com.mongodb.client.model.expressions.MqlUnchecked.Unchecked.PRESENT;
 
+/**
+ * A map {@link Expression value} in the context of the MongoDB Query
+ * Language (MQL). An map is a finite, unordered a set of
```

Comment:
1. "unordered" - I don't think we should say that it's unordered. Having the entries ordered in a map is more important than having the fields ordered in a document (and the document type is documented as being ordered): there a user has no way of observing the order until a document is converted to a map (the user still may influence the order via the document API).
2. "unordered a set" -> "a set"? (though, given #1 this is irrelevant).

Reply:
Used "A map is a finite set of..."

Reply:
My understanding is that you still want to not express that map entries are ordered. The order of map entries is an important property that is already there and we need good reasons to deprive users of it. What are those reasons?

Reply:
We discussed this. The proposal was to effectively use the wording at the class level:

> Despite a document in MongoDB being an ordered set of fields, users must not rely on this guarantee (or property) when using DocumentExpression/MapExpression."

I put a shorter variant of this on `entrySet`, in line with the docs on `distinct`. I could see no way for users to rely on this property via our API using Docs, and only entrySet can rely on it for Maps. (The behaviour of storing/fetching is outside of the scope of our API, though we try to maintain/round-trip properties.) Prominently mentioning it seems to give the misleading impression that this is a bigger concern that it is in practice.